syntax = "proto3";

package vc;

service LipSyncer {
    rpc facedetect(stream FaceBBRequest) returns (EmbedResponse) {}
    rpc lipsync(stream LipSyncRequest) returns (stream LipSyncResponse) {}
}

message FaceDetectRequest {
    bytes image = 1;
}

message FaceDetectResponse {
    string embed_id = 1;
}

message LipSyncRequest {
    bytes buffer = 1;
}

message LipSyncResponse {
    bytes image = 1;
}
